src/key_value_store/KeyValueStoreInterface.h

```diff
@@ -15,6 +16,8 @@ class KeyValueStoreInterface {
 
   virtual bool Put(const std::string &key, const std::string &value) = 0;
   virtual std::vector<std::string> Get(const std::string &key) = 0;
+  virtual bool Get(const std::string &tag,
+                   std::function<bool(std::string)>& callback) = 0;
```

Comment:
I agree with you, but since I am still not completely sure how the callback in the KV Client will be used I was thinking of leaving it as non-const until the end.  What if I just add a TODO above this and make it const if the Client needs to modify?  I might also even make the argument `std::string` const and by reference as well after.